libcextract/SymbolExternalizer.cpp

```diff
@@ -71,142 +87,161 @@ using namespace llvm;
+         * private symbol.
+         */
+        if (SE.Ibt)
+          replacement = new_decl->getName().str();
 
-  /** A reference to SymbolExternalizer.  */
-  SymbolExternalizer &SE;
+        sym->Done = true;
+        sym->Wrap = !SE.Ibt;
+      } else {
+        /* If we externalized this function, then all further delcarations of
+           this function shall be discarded.  */
 
-  /** The new variable declaration to replace the to be externalized function.  */
-  ValueDecl *NewSymbolDecl;
+        /* Get source location of old function declaration.  */
+        SE.Remove_Text(decl->getSourceRange(), 1000);
+      }
+    } else if (type == ExternalizationType::WEAK) {
+      /* Now checks if this is a function or a variable delcaration.  */
+      if (FunctionDecl *func = dyn_cast<FunctionDecl>(decl)) {
+        /* In the case it is a function we need to remove its declaration that
+           have a body.  */
+        if (func->hasBody()) {
+          FunctionDecl *with_body = func->getDefinition();
+          if (with_body != func)
+            SE.Remove_Text(with_body->getSourceRange(), 1000);
+        }
+      }
+    } else if (type == ExternalizationType::RENAME) {
+      /* Get SourceRange where the function identifier is.  */
+      auto ids = Get_Range_Of_Identifier_In_SrcRange(decl->getSourceRange(),
+                                                     decl->getName());
+      assert(ids.size() > 0 && "Decl name do not match required identifier?");
+
+      SourceRange id_range = ids[0];
+      std::string new_name = RENAME_PREFIX + decl->getName().str();
+      sym->new_name = new_name;
+      if (!sym->Done) {
+        /* Only register the first decl rename of the same variable.  */
+        SE.Log.push_back({.OldName = decl->getName().str(),
+                       .NewName = new_name,
+                       .Type = type});
+        sym->Done = true;
+      }
 
-  /** Name of the to be replaced function.  */
-  const std::string &OldSymbolName;
+      /* In the case there is a `static` modifier in function, try to drop it.  */
+      if (FunctionDecl *fdecl = dyn_cast<FunctionDecl>(decl)) {
+        SE.Drop_Static(fdecl);
+      }
 
-  bool Wrap;
-};
+      /* Replace text content of old declaration.  */
+      SE.Replace_Text(id_range, new_name, 100);
 
-static std::vector<SourceRange>
-Get_Range_Of_Identifier_In_SrcRange(const SourceRange &range, const char *id)
-{
-  std::vector<SourceRange> ret = {};
-  StringRef string = PrettyPrint::Get_Source_Text(range);
+      sym->Wrap = false;
+    }
 
-  /* Tokenize away the function-like macro stuff or expression, we only want
-     the identifier.  */
-  const char *token_vector = " ().,;+-*/^|&{}[]<>^&|\r\n\t";
+    return VISITOR_CONTINUE;
+  }
 
-  /* Create temporary buff, strtok modifies it.  */
-  unsigned len = string.size();
-  char buf[len + 1];
-  memcpy(buf, string.data(), len);
-  buf[len] = '\0';
+  bool VisitDeclRefExpr(DeclRefExpr *expr)
```

Comment:
Please explain what this method is supposed to do. Something in the lines of "Update the use of the externalize variables so it references our new symbol."